server/src/main/java/org/opensearch/gateway/remote/ClusterMetadataManifest.java

```diff
@@ -246,6 +311,10 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeCollection(indices);
         out.writeString(previousClusterUUID);
         out.writeBoolean(clusterUUIDCommitted);
+        if (out.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
should this also be based on codec version now?

Reply:
No, here we would need version based check only. This will be mainly for node to node communication, where we check the StreamInput/Output's version.